From 560baf686515dcacd94dbc26895a974048dcc308 Mon Sep 17 00:00:00 2001
From: Hyungchan Kim <hyungchan2.kim@lge.com>
Date: Wed, 26 May 2021 17:45:05 +0900
Subject: [PATCH] Fix compile error with g++ 7.5.0

This fixes a few compilation errors in a host that has g++ 7.5.0.
There is no fix from upstream as 8.1 is the minimum gcc version of Qt.

Upstream-Status : Inappropriate [workaround]
---
 src/corelib/kernel/qproperty_p.h | 2 +-
 src/corelib/time/qdatetime.cpp   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/kernel/qproperty_p.h b/src/corelib/kernel/qproperty_p.h
index 71f4f9f5e9..1bf3d490f1 100644
--- a/src/corelib/kernel/qproperty_p.h
+++ b/src/corelib/kernel/qproperty_p.h
@@ -551,7 +551,7 @@ public:
             if (!inBindingWrapper(storage))
                 notify(bd);
         }
-        if constexpr (Signal != nullptr) {
+        if constexpr (!std::is_same_v<decltype(Signal), nullptr_t>) {
             if constexpr (SignalTakesValue::value)
                 (owner()->*Signal)(value());
             else
diff --git a/src/corelib/time/qdatetime.cpp b/src/corelib/time/qdatetime.cpp
index f0841ca853..98e20ae921 100644
--- a/src/corelib/time/qdatetime.cpp
+++ b/src/corelib/time/qdatetime.cpp
@@ -4000,7 +4000,7 @@ void QDateTime::setMSecsSinceEpoch(qint64 msecs)
         status |= QDateTimePrivate::ValidWhenMask;
         break;
     case Qt::OffsetFromUTC:
-        if (!add_overflow(msecs, d->m_offsetFromUtc * MSECS_PER_SEC, &msecs))
+        if (!add_overflow(msecs, qint64(d->m_offsetFromUtc * MSECS_PER_SEC), &msecs))
             status |= QDateTimePrivate::ValidWhenMask;
         break;
     case Qt::TimeZone:
